index.js

```diff
@@ -159,4 +159,51 @@ mung.headersAsync = function headersAsync (fn) {
     }
 }
 
+mung.writeJson = function writeJson (fn, options = {}) {
+    return function (req, res, next) {
+        const original = res.write;
+        const mungError = options.mungError;
+
+        res.write = (body, ...args) => {
+            const contentType = res.get('Content-Type');
+
+            if (res.headersSent) {
+                return;
+            }
+
+            // Do not mung on errors
+            if (!mungError && res.statusCode >= 400) {
+                return original.call(res, body);
+            }
+
+            // If response type is not application/json,
+            // just call the original res.write function
+            if (!(contentType && ~contentType.indexOf('application/json'))) {
+                return original.call(res, body);
+            }
+
+            try {
+                const json = JSON.parse(body);
+                const modifiedJSON = fn(json, req, res);
+
+                if (res.headersSent) {
+                    return;
+                }
+
+                // If null, then 204 No Content
+                if (!modifiedJSON) {
+                    return res.status(204).end();
+                }
+
+                res.set('Content-Length', Buffer.byteLength(JSON.stringify(modifiedJSON), 'utf8'));
```

Comment:
What about cases when charset other than `utf8` is used?

Reply:
Good catch. I can add `encoding` as an option.